#! /bin/sh /usr/share/dpatch/dpatch-run
## 08_fix_cxx_includes.dpatch by Andrew Ross <andrewross@users.sourceforge.net>
##
## DP: Add in missing include files for C++ examples to fix compile errors
## DP: with gcc 4.3. (This patch was taken from the upstream SVN repository.)

@DPATCH@

--- plplot-5.8.0.orig/examples/c++/plc++demos.h	(revision 8091)
+++ plplot-5.8.0/examples/c++/plc++demos.h	(working copy)
@@ -8,7 +8,14 @@
 
 /* common includes */
 #include "plstream.h"
+
+/* plplot actually includes the C versions of these 2, but
+ * add here for completeness */
+#include <cstdio>
+#include <cstdlib>
+
 #include <iostream>
+#include <cstring>
 #include <cmath>
 
 /* common defines */
--- plplot-5.8.0.orig/examples/c++/x21.cc	(revision 8092)
+++ plplot-5.8.0/examples/c++/x21.cc	(working copy)
@@ -27,7 +27,7 @@
 //---------------------------------------------------------------------------//
 
 #include "plc++demos.h"
-#include <time.h>
+#include <ctime>
 
 #if !defined(HAVE_ISNAN)
   #define isnan(x) ((x) != (x))
